server/src/main/java/org/opensearch/telemetry/tracing/ThreadContextBasedTracerContextStorage.java

```diff
@@ -52,17 +53,24 @@ public void put(String key, Span span) {
     @Override
     public Map<String, Object> transients(Map<String, Object> source) {
         final Map<String, Object> transients = new HashMap<>();
-
         if (source.containsKey(CURRENT_SPAN)) {
             final SpanReference current = (SpanReference) source.get(CURRENT_SPAN);
             if (current != null) {
                 transients.put(CURRENT_SPAN, new SpanReference(current.getSpan()));
             }
         }
-
         return transients;
     }
 
+    @Override
+    public Map<String, Object> transients(Map<String, Object> source, boolean isSystemContext) {
```

Comment:
Same for `headers`?

Reply:
Header we use to propagate context from one node to another so, I don't think we need any systemContext related special behaviour.

Reply:
I mean we introduced this method which we should implement, others should be deprecated
```
default Map<String, String> headers(Map<String, Object> source, boolean isSystemContext) {
```

Reply:
Yes, that make sense.